core/trino-main/src/main/java/io/trino/operator/output/PagePartitioner.java

```diff
@@ -477,6 +473,12 @@ private Page getPartitionFunctionArguments(Page page)
     }
 
     public void flush(boolean force)
+    {
+        flushColumnarBuffers(force);
```

Comment:
technically, `BlockBuilder` can also be constructed in a columnar way and is storing values in columnar way.

Maybe:
```
flushPositionsAppenders
flushPageBuilders
```
and rename `columnarPageBuilders` to just `positionsAppenders`